sdk/search/azure-search/src/main/java/com/azure/search/SearchIndexClientBuilder.java

```diff
@@ -34,81 +33,131 @@
 public class SearchIndexClientBuilder {
-
-    // This header tells the server to return the request id in the HTTP response. Useful for correlation with what
-    // request was sent.
+    /*
+     * This header tells the service to return the request ID in the HTTP response. This is useful for correlating the
+     * request sent to the response.
+     */
     private static final String ECHO_REQUEST_ID_HEADER = "x-ms-return-client-request-id";
 
+    /*
+     * This is the name of the request ID header that the Search service expects. If the customer doesn't pass
+     * RequestOptions into a service call this will allow for the client to implicitly add a request ID header,
+     * otherwise the passed request ID will be used.
+     */
+    private static final String CLIENT_REQUEST_ID = "client-request-id";
+
     private static final String SEARCH_PROPERTIES = "azure-search.properties";
     private static final String NAME = "name";
     private static final String VERSION = "version";
 
-    SearchApiKeyCredential searchApiKeyCredential;
-    SearchServiceVersion apiVersion;
-    String endpoint;
-    HttpClient httpClient;
-    HttpLogOptions httpLogOptions;
-    Configuration configuration;
-    List<HttpPipelinePolicy> policies;
-    private String clientName;
-    private String clientVersion;
+    private final ClientLogger logger = new ClientLogger(SearchIndexClientBuilder.class);
+    private final List<HttpPipelinePolicy> policies = new ArrayList<>();
+    private final HttpHeaders headers = new HttpHeaders().put(ECHO_REQUEST_ID_HEADER, "true");
 
+    private final String clientName;
+    private final String clientVersion;
+
+    private SearchApiKeyCredential searchApiKeyCredential;
+    private SearchServiceVersion apiVersion;
+    private String endpoint;
+    private HttpClient httpClient;
+    private HttpPipeline httpPipeline;
+    private HttpLogOptions httpLogOptions;
+    private Configuration configuration;
     private String indexName;
-    private final HttpHeaders headers;
     private RetryPolicy retryPolicy;
-    private final ClientLogger logger = new ClientLogger(SearchIndexClientBuilder.class);
+
 
     /**
-     * Default Constructor
+     * Creates a builder instance that is able to configure and construct {@link SearchIndexClient SearchIndexClients}
+     * and {@link SearchIndexAsyncClient SearchIndexAsyncClients}.
      */
     public SearchIndexClientBuilder() {
-        apiVersion = SearchServiceVersion.getLatest();
-        policies = new ArrayList<>();
-        httpClient = HttpClient.createDefault();
-        httpLogOptions = new HttpLogOptions();
-
         Map<String, String> properties = CoreUtils.getProperties(SEARCH_PROPERTIES);
         clientName = properties.getOrDefault(NAME, "UnknownName");
         clientVersion = properties.getOrDefault(VERSION, "UnknownVersion");
-        headers = new HttpHeaders()
```

Comment:
Why removed the default value?

Reply:
This was inline instantiated at the property, no need to include this in the constructor if it will be created for every client.